### PR TITLE
Qubit `create_initial_state` added to the devices module

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
The new python simulator will need a way to create initial states, and this will be that way.

**Description of the Change:**
Adds the `devices/qubit` module, including the `create_initial_state` function. It optionally accepts a prep operation (implementing the `StatePrep` interface), as well as a `like=interface` option. Note that if a prep operator AND a `like` option is given, only the operator is considered (and the interface is inferred by the operator itself).

**Benefits:**
Circuits will be able to initialize the state programatically and functionally in the new Python simulator device!

**Possible Drawbacks:**
None

### CODE REVIEW

File: pennylane/devices/qubit/initialize_state.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from typing import Iterable, Union
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+def create_initial_state(
+    wires: Union[qml.wires.Wires, Iterable],
+    prep_operation: qml.operation.StatePrep = None,
+    like: str = "autograd",
```
Comment (albi3ro) [inline pennylane/devices/qubit/initialize_state.py:25]: Probably best to start with `"numpy"`.

Comment (timmysilv) [reply pennylane/devices/qubit/initialize_state.py:25]: my intention was to [do the same as QNode](https://github.com/PennyLaneAI/pennylane/blob/07b3e6e3cf9ec05c5f7e35bcb784825ccb04ca6d/pennylane/qnode.py#L355), but I'm happy to use `numpy` if we're ready to start moving away from autograd.

File: tests/devices/qubit/test_initialize_state.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        state = create_initial_state([0, 1, 2], prep_operation=prep_op)
+        assert state[0, 1, 0] == 1
+        state[0, 1, 0] = 0  # set to zero to make test below simple
+        assert qml.math.allequal(state, np.zeros((2, 2, 2)))
+
+    def test_create_initialize_state_prefers_op_interface_over_like(self):
+        """Tests that the like argument is ignored when a prep-op is provided."""
+        prep_op = self.DefaultPrep([0, 0, 0, 1], wires=[0, 1])
+        state = create_initial_state([0, 1], prep_operation=prep_op, like="torch")
+        assert qml.math.get_interface(state) == "numpy"
```
Comment (albi3ro) [inline tests/devices/qubit/test_initialize_state.py:65]: We should actually prefer `like` over prep-op interface, especially if the prep op is `numpy`.


Take for consideration `QuantumScript([qml.RX(torch.tensor(1.0), 0)], [qml.expval(qml.PauliZ(0))], prep_op)`.  WE will need the returned state to be `torch` or we will get errors whe applying the rotation gate.

**Review by albi3ro — APPROVED** [review]
Looks good to me 👍

**Pull request merged**